lib/Serialization/DeserializeSIL.cpp

```diff
@@ -1436,7 +1444,25 @@ bool SILDeserializer::readSILInstruction(SILFunction *Fn, SILBasicBlock *BB,
   }
   // SWIFT_ENABLE_TENSORFLOW
   case SILInstructionKind::GraphOperationInst: {
-    llvm_unreachable("Unimplemented");
+    auto EndOfArgValues = 3 * NumArguments;
+    Identifier MangledName = MF->getIdentifier(ValID);
+    SmallVector<SILValue, 4> Args;
+    for (unsigned i = 0, e = EndOfArgValues; i < e; i += 3) {
+      auto ArgASTTy = MF->getType(ListOfValues[i+1]);
+      auto ArgTy = getSILType(ArgASTTy,
+                              (SILValueCategory)(unsigned)ListOfValues[i+2]);
+      Args.push_back(getLocalValue(ListOfValues[i], ArgTy));
+    }
+    SmallVector<SILType, 4> ResultSILTypes;
+    for (unsigned i = EndOfArgValues, e = ListOfValues.size(); i < e; i += 2) {
+      auto ASTTy = MF->getType(ListOfValues[i]);
+      auto Ty = getSILType(ASTTy,
+                           (SILValueCategory)(unsigned)ListOfValues[i+1]);
+      ResultSILTypes.push_back(Ty);
+    }
+    ResultVal = Builder.createGraphOperation(Loc, MangledName, Args, {},
```

Comment:
Is it not deserializing attributes? If it's not handled yet, please add a TODO somewhere in both serialization and deserialization.

Reply:
I believe the second loop deserializes attributes. It would be good to add some comments if that is the case.

Reply:
It does not do attributes. The second loop does result types. I will add TODOs.